CorsixTH/Lua/world.lua

```diff
@@ -299,69 +288,11 @@ function World:toggleInformation()
   self.room_information_dialogs = not self.room_information_dialogs
 end
 
---! Load goals to win and lose from the map, and store them in 'self.goals'.
---! Also set 'self.winning_goal_count'.
+--! In EndConditions, collect and test the win and lose conditions of the level.
 function World:determineWinningConditions()
-  local winning_goal_count = 0
-  -- No conditions if in free build mode!
-  if self.free_build_mode then
-    self.goals = {}
-    self.winning_goal_count = winning_goal_count
-    return
-  end
-  -- Determine winning and losing conditions
-  local world_goals = {}
-
-  -- There might be no winning criteria (i.e. the demo), then
-  -- we don't have to worry about the progress report dialog
-  -- since it doesn't exist anyway.
-  local win = self.map.level_config.win_criteria
-  if win then
-    for _, values in pairs(win) do
-      if values.Criteria ~= 0 then
-        winning_goal_count = winning_goal_count + 1
-        local crit_name = self.level_criteria[values.Criteria].name
-        world_goals[crit_name] = {
-          name = crit_name,
-          win_value = values.Value,
-          boundary = values.Bound,
-          criterion = values.Criteria,
-          max_min_win = values.MaxMin,
-          group = values.Group,
-          number = winning_goal_count,
-        }
-        world_goals[#world_goals + 1] = world_goals[crit_name]
-      end
-    end
-  end
-  -- Likewise there might be no losing criteria (i.e. the demo)
-  local lose = self.map.level_config.lose_criteria
-  if lose then
-    for _, values in pairs(lose) do
-      if values.Criteria ~= 0 then
-        local crit_name = self.level_criteria[values.Criteria].name
-        if not world_goals[crit_name] then
-          world_goals[crit_name] = {number = #world_goals + 1, name = crit_name}
-          world_goals[#world_goals + 1] = world_goals[crit_name]
-        end
-        world_goals[crit_name].lose_value = values.Value
-        world_goals[crit_name].boundary = values.Bound
-        world_goals[crit_name].criterion = values.Criteria
-        world_goals[crit_name].max_min_lose = values.MaxMin
-        world_goals[crit_name].group = values.Group
-        world_goals[world_goals[crit_name].number].lose_value = values.Value
-        world_goals[world_goals[crit_name].number].boundary = values.Bound
-        world_goals[world_goals[crit_name].number].criterion = values.Criteria
-        world_goals[world_goals[crit_name].number].max_min_lose = values.MaxMin
-        world_goals[world_goals[crit_name].number].group = values.Group
-      end
-    end
+  if not (self.app.using_demo_files or self.free_build_mode) then
```

Comment:
The demo definitely has end conditions does it not?
Edit: Looks like CorsixTH actually loads the base_config, which give no win/lose condition so this isn't wrong.
But a different issue for later is that the Demo has a defined endCondition in the readme:
```txt
You have to cure 20 patients in a year in this demo so look after your
patients well and provide for their every need.
```
And in the demo itself from DOSBOX
![image](https://github.com/CorsixTH/CorsixTH/assets/20030128/f80942db-e112-47c9-81c5-eca6866e9ac7)